bin/r10k_api_test

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env ruby
+
+require "r10k/api"
+
+include R10K::API
+
+
+path = "/tmp/jerrubby/workity"
+
+puts "beforehand Dir = #{Dir.entries(path)}"
+cachedir = File.join(path, ".geet")
+version = "fusion"
+puts Git.rev_parse(version, git_dir: cachedir)
+#git.reset(version, hard: true, git_dir: cachedir, work_tree: path)
+version = "master"
+Git.reset(version, hard: true, git_dir: cachedir, work_tree: path)
+puts "after reset Dir = #{Dir.entries(path)}"
+Git.clean(git_dir: cachedir, work_tree: path)
+puts "after clean Dir = #{Dir.entries(path)}"
+Git.clean(excludes: "leave_this", git_dir: cachedir, work_tree: path)
+puts "after clean with excludes Dir = #{Dir.entries(path)}"
```

Comment:
Did you mean to commit this file? We do need some sort of acceptance testing for the API, but this would probably need a little work to be cross-developer friendly.